repositories.md

```diff
@@ -5,9 +5,73 @@ css_id: repositories
+
+* Rotation of the The Time Server's key is performed by listing the new key in the Director's Root metadata, in the same manner as other role keys are listed, and also in the custom field of the Director Repository's Targets metadata (for partial verification secondaries).
+
+#### Changes to the Director repository
+If a Time Server is in use, a representation of the Time Server public key is CONDITIONALLY REQUIRED in Director repository root metadata.
+
+If a Time Server is implemented AND partial-verification secondaries are used, the following metadata is CONDITIONALLY REQUIRED in the Director repository's Targets metadata:
+
+* A representation of the public key(s) for the Time Server, similar to the representation of public keys in Root metadata.
+
+Listing the public key of the Time Server in Director targets metadata is necessary to allow partial-verification secondaries to perform time server key rotation.
+
+#### Changes to a Primary
+
+If the Time Server is implemented, the primary is CONDITIONALLY REQUIRED to use the following procedure to verify the time. This procedure occurs after the vehicle version manifest is sent and will fulfill the ["Download and check current time"](https://uptane.github.io/papers/ieee-isto-6100.1.0.0.uptane-standard.html#check_time_primary) step of the Uptane Standard.
+
+1. Gather the tokens from each secondary ECU's version report.
+2. Send the list of tokens to the Time Server to fetch the current time. The Time Server responds as described in [Time Server](#time_server), providing a cryptographic attestation of the last known time.
+3. If the Time Server's response meets the criteria below, update the primary ECU's clock and retain the Time Server's response for distribution to secondary ECUs, otherwise discard it and proceed without an updated time.  The criteria for checking the Time Server's response are:
+  - The signature over the Time Server's response is valid.
+  - All the tokens provided to the Time Server have been included in the response.
+  - The time in the Time Server's response is later than the last time verified in this manner.
+
+#### ECU Version Report
+
+The payload of the ECU version report sent to the Director might contain the token sent to the Time Server. This token is in the version report sent from secondaries to the primary, and so is in the signed version of the version report. If the token is removed, the signature will not match.
+
+#### Changes to all ECUs
+
+At build time, ECUs will be provisioned with an attestation of the current time downloaded from the Time Server.
+
+As the first step to verifying metadata, described as ["Load and verify the current time or the most recent securely attested time"](https://uptane.github.io/papers/ieee-isto-6100.1.0.0.uptane-standard.html#partial_verification) in the Standard, the ECU SHOULD load and verify the most recent time from the Time Server using the following procedure:
```

Comment:
Actually I think you mean [this](https://uptane.github.io/papers/ieee-isto-6100.1.0.0.uptane-standard.html#check_time_primary) and [this](https://uptane.github.io/papers/ieee-isto-6100.1.0.0.uptane-standard.html#verify_time)